cmd/platform/export.go

```diff
@@ -1,39 +1,205 @@
+			l := logger.Get()
+
+			l.Debug().Msgf("Export Subcommand Called.")
+
+			apiClient, err := initApiClient(cmd.Context(), cmd)
 			if err != nil {
 				output.Format(cmd, output.CommandOutput{
-					Message: "Export failed.",
-					Fatal:   err,
+					Message: "Unable to initialize PingOne SDK client",
 					Result:  output.ENUMCOMMANDOUTPUTRESULT_FAILURE,
 				})
+				return err
 			}
+
+			// Using the --service parameter(s) provided by user, build list of connectors to export
+			exportableConnectors := []connector.Exportable{}
+			for _, service := range services {
+				switch service {
+				case pingone_platform.ServiceName:
+					exportableConnectors = append(exportableConnectors, pingone_platform.Connector(cmd.Context(), apiClient, viper.GetString(pingoneWorkerEnvironmentIdParamConfigKey)))
+				default:
+					output.Format(cmd, output.CommandOutput{
+						Message: fmt.Sprintf("Provided service not recognized: %s", service),
+						Result:  output.ENUMCOMMANDOUTPUTRESULT_FAILURE,
+					})
+					return err
+				}
+			}
+
+			// Select export format based on user's --export-format parameter
+			var connectorExportFormat string
+			switch exportFormat {
+			case connector.ENUMEXPORTFORMAT_HCL:
+				connectorExportFormat = connector.ENUMEXPORTFORMAT_HCL
+			default:
+				output.Format(cmd, output.CommandOutput{
+					Message: fmt.Sprintf("Provided export format not recognized: %s. Defaulting to HCL export format.", exportFormat),
+					Result:  output.ENUMCOMMANDOUTPUTRESULT_NOACTION_WARN,
+				})
+				connectorExportFormat = connector.ENUMEXPORTFORMAT_HCL
+			}
+
+			// Loop through user defined exportable connectors and export them
+			for _, connector := range exportableConnectors {
+				l.Debug().Msgf("Exporting %s service...", connector.ConnectorServiceName())
+
+				err := connector.Export(connectorExportFormat, outputDir, overwriteExport)
+				if err != nil {
+					output.Format(cmd, output.CommandOutput{
+						Message: fmt.Sprintf("Export failed for service: %s.", connector.ConnectorServiceName()),
+						Result:  output.ENUMCOMMANDOUTPUTRESULT_FAILURE,
+					})
+					return err
+				}
+			}
+			return nil
 		},
 	}
 
+	// Add flags that are not tracked in the viper configuration file
+	cmd.Flags().StringVar(&exportFormat, "export-format", "HCL", "Specifies export format\nValid options: 'HCL'")
+	cmd.Flags().StringSliceVar(&services, "service", []string{"pingone-platform"}, "Specifies service(s) to export")
+	cmd.Flags().StringVar(&outputDir, "output-directory", "", "Specifies output directory for export (Default: Present working directory)")
+	cmd.Flags().BoolVar(&overwriteExport, "overwrite", false, "Overwrite existing generated exports if set.")
+
+	// Add flags that are bound to configuration file keys
+	cmd.Flags().String(pingoneWorkerEnvironmentIdParamName, os.Getenv("PINGONE_ENVIRONMENT_ID"), "The ID of the PingOne environment that contains the worker token client used to authenticate.")
```

Comment:
We should include the corresponding environment variables in these descriptions so users know they can use them instead.

Reply:
Added!